imap_processing/cli.py

```diff
@@ -1232,36 +1222,33 @@ def do_processing(
             The list of processed products.
         """
         print(f"Processing Spacecraft {self.data_level}")
-
+        processed_dataset = []
         if self.data_level == "l1a":
-            # File path is expected output file path
-            input_files = dependencies.get_file_paths(source="spacecraft")
-            if len(input_files) > 1:
-                raise ValueError(
-                    f"Unexpected dependencies found for Spacecraft L1A: "
-                    f"{input_files}. Expected only one dependency."
+            if self.descriptor == "quaternions":
+                # File path is expected output file path
+                input_files = dependencies.get_file_paths(source="spacecraft")
+                if len(input_files) > 1:
+                    raise ValueError(
+                        f"Unexpected dependencies found for Spacecraft L1A: "
+                        f"{input_files}. Expected only one dependency."
+                    )
+                datasets = list(quaternions.process_quaternions(input_files[0]))
+                processed_dataset.extend(datasets)
+            elif self.descriptor == "pointing-attitude":
+                spice_inputs = dependencies.get_file_paths(
+                    data_type=SPICESource.SPICE.value
                 )
-            datasets = list(quaternions.process_quaternions(input_files[0]))
-            return datasets
-        elif self.data_level == "spice":
-            spice_inputs = dependencies.get_file_paths(
-                data_type=SPICESource.SPICE.value
-            )
-            ah_paths = [path for path in spice_inputs if ".ah" in path.suffixes]
-            if len(ah_paths) != 1:
-                raise ValueError(
-                    f"Unexpected spice dependencies found for Spacecraft "
-                    f"pointing_kernel: {ah_paths}. Expected exactly one "
-                    f"attitude history file."
+                ah_paths = [path for path in spice_inputs if ".ah" in path.suffixes]
+                # TODO: refactor if multiple attitude history files are found
+                pointing_kernel_paths = (
+                    pointing_frame.generate_pointing_attitude_kernel(ah_paths[0])
```

Comment:
I think that it is OK to get rid of the `raise ValueError` but we want to pass the last ah file path to the job.
```suggestion
                    pointing_frame.generate_pointing_attitude_kernel(ah_paths[-1])
```

Reply:
yep. That fixed the concerns. Thank you!